src/index.js

```diff
@@ -336,7 +336,8 @@ function removeNodeFromHost(host, node) {
 }
 
 function removeSlotFromRoot(root, node) {
-  node.assignedNodes().forEach(slotNodeFromSlot);
+  const assignedNodes = Array.prototype.slice.call(node.assignedNodes());
```

Comment:
I'm not really very worried because for this to become a performance bottleneck, you'd need to be doing a lot of adding / removing of slot nodes, which I think is unlikely.